src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -118,7 +121,7 @@ public static string GetFullPath(string path, string basePath)
             // to GetFullPath() won't do anything by design. Additionally, GetFullPathName() in
             // Windows doesn't root them properly. As such we need to manually remove segments.
             return PathInternal.IsDevice(combinedPath)
-                ? RemoveRelativeSegments(combinedPath, PathInternal.GetRootLength(combinedPath))
+                ? RemoveRelativeSegments(combinedPath, PathInternal.GetRootLength(combinedPath) - 1)
```

Comment:
Why remove one from root length? This needs comments.

Reply:
For cases like where combined root = \\.\?\C:\.\tmp

Reply:
Without a detailed comment this isn't understandable. Something like "Paths at this point are in the form of ... we skip to the last character of the root when calling RemoveRelativeSegments so that ...".  What are all of the states the path root can be in at this point? `\\?\`, `\\?\foo`, `\\?\foo\`? As complicated and subtle as this stuff is, don't hesitate to put very verbose comments in. Even I can't follow this one without a comment.

Reply:
Also is " on end